File: 01_projects/code/python-playwright/cbaps-dawms/robust/dawms/pages/signature_routing_page.py
from playwright.sync_api import Page
from loguru import logger

class SignatureRoutingPage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.signer_dropdown = page.locator("#signer")
        self.submit_btn = page.locator("button:has-text('Submit for Signature')")
        self.pw.wait_for_selector("#signer")
    
    def submit_for_signature(self, signer: str):
        from .milestone_status_page import MilestoneStatusPage
        logger.info(f"✍️ Submitting: {signer}")
        self.pw.select_dropdown(self.signer_dropdown, signer)
        self.pw.click_element(self.submit_btn)
        return MilestoneStatusPage(self.page, self.pw)

File: 01_projects/code/java-playwright/cbaps-dawms/robust/src/main/java/com/playwright/dawms/pages/SignatureRoutingPage.java
package com.playwright.dawms.pages;
import com.microsoft.playwright.*;
import com.playwright.cbaps.library.EnhancedPlaywrightManager;

public class SignatureRoutingPage {
    private Page page;
    private EnhancedPlaywrightManager pwm;
    private Locator signerDropdown, submitButton;
    
    public SignatureRoutingPage(Page page, EnhancedPlaywrightManager pwm) {
        this.page = page; this.pwm = pwm;
        this.signerDropdown = page.locator("#signer");
        this.submitButton = page.locator("button:has-text('Submit for Signature')");
        pwm.waitUntilElementVisible("#signer");
    }
    
    public MilestoneStatusPage submitForSignature(String signer) {
        pwm.selectDropdown(signerDropdown, signer);
        pwm.clickElement(submitButton);
        return new MilestoneStatusPage(page, pwm);
    }
}
